### PR TITLE
Added optional tags/dimensions for PartitionId/ReplicaId as addition/…

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Added optional tags/dimensions for PartitionId/ReplicaId as addition/alternative to ServiceAddress for rntbd/request-level metrics

Sample:
```java
new CosmosMicrometerMetricsOptions()
            .meterRegistry(this.meterRegistry)
            .setMetricCategories(metricCategories)
            .configureDefaultTagNames(
                CosmosMetricTagName.DEFAULT,
                CosmosMetricTagName.PARTITION_ID,
                CosmosMetricTagName.REPLICA_ID);
```

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #### Features Added
 * Added the capability to filter request-level metrics based on diagnostic thresholds. Request-level metrics usually are used to capture metrics per backend endpoint/replica - a high cardinality dimension. Filtering by diagnostic thresholds reduces the overhead - but also means request-level metrics can only be used for debugging purposes - not for monitoring purposes. So, it is important to use the unfiltered operation-level metrics for health monitoring in this case. - See [PR 35114](https://github.com/Azure/azure-sdk-for-java/pull/35114)
-
+* Added optional tags/dimensions for PartitionId/ReplicaId as alternative to ServiceAddress for direct-mode (rntbd) request-level metrics. - See [PR 35164](https://github.com/Azure/azure-sdk-for-java/pull/35164)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:7]: why the ServiceAddress is not enough?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java [hunk 2]
```diff
@@ -650,10 +650,39 @@ private Tags createRequestTags(
...
+            boolean containsPartitionId = metricTagNames.contains(TagName.PartitionId);
+            boolean containsReplicaId = metricTagNames.contains(TagName.ReplicaId);
+            if (containsPartitionId || containsReplicaId) {
+
+                String partitionId = "NONE";
+                String replicaId = "NONE";
+
+                if (serviceAddress != null) {
+                    String[] serviceAddressParts = effectiveServiceAddress.split("/");
+                    if (serviceAddressParts.length == 9) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java:669]: Can we put a sample here - to show how this serviceAddressParts array looks like (after splitting or pre-splitting) - probably easier to understand and debug the code if needed.

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @FabianMeiswinkel

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md:7]: I think WMT preferred just partitionId as metric - shorter when looking at graphs - and consuming less space in the metrics store.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java:671]: Done

**Pull request merged**